src/bin/chronos-splits-editor/main.rs

```diff
@@ -0,0 +1,646 @@
+use clap::Parser;
+use cursive::align::HAlign;
+use cursive::direction::Orientation;
+use cursive::reexports::enumset::EnumSet;
+use cursive::theme::ColorStyle;
+use cursive::traits::*;
+use cursive::view::SizeConstraint::*;
+use cursive::views::*;
+use cursive::Cursive;
+use livesplit_core::run::{
+    editor::{SegmentState, SelectionState},
+    parser::composite,
+    saver::livesplit::{self, IoWrite},
+    Editor,
+};
+use std::error::Error;
+use std::fs::{self, File};
+use std::path::Path;
+
+use livesplit_core::timing::formatter::{
+    none_wrapper::{EmptyWrapper, NoneWrapper},
+    Accuracy, SegmentTime, TimeFormatter,
+};
+
+#[derive(Parser)]
+struct Args {
+    splits_file: String,
+}
+
+pub struct GlobalState {
+    splits_editor: Editor,
+    splits_file: String,
+}
+
+// this whole thing is a mess because cursive has some of the most atrocious
+// state management i have ever seen
+//
+// this will eventually be rewritten to use either dioxus or intuitive
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+    let path = Path::new(&args.splits_file);
+    let splits = match fs::read(path) {
+        Ok(file) => {
+            let parsed = composite::parse(&file, None, false)
+                .expect("Not a valid splits file");
+            parsed.run
+        }
+        Err(_) => {
+            let default = chronos::DEFAULT_SPLITS;
+            let parsed = composite::parse(&default, None, false)
+                .expect("Not a valid splits file");
+            parsed.run
+        }
+    };
+
+    let editor = Editor::new(splits)?;
+
+    let globals = GlobalState {
+        splits_editor: editor,
+        splits_file: args.splits_file,
+    };
+
+    let backend = {
+        let backend = cursive::backends::crossterm::Backend::init()?;
+        let buffered = cursive_buffered_backend::BufferedBackend::new(backend);
+        Box::new(buffered)
+    };
+
+    let mut siv = cursive::CursiveRunner::new(Cursive::new(), backend);
+    cursive::logger::init();
+    siv.load_toml(chronos::CURSIVE_THEME).unwrap();
+
+    siv.add_global_callback('~', cursive::Cursive::toggle_debug_console);
+    siv.set_user_data(globals);
+    main_splits(&mut siv);
+    siv.run();
+    Ok(())
+}
+
+#[rustfmt::skip]
```

Comment:
please avoid suppressing rustfmt everywhere

Reply:
This is mostly because of PaddedView going vertical with its params - I'll look at a cleaner solution (ex: multiple assignment)